src/Api/Monetization/Controller/CompanyReportDefinitionControllerInterface.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Apigee\Edge\Api\Monetization\Controller;
+
+interface CompanyReportDefinitionControllerInterface extends ReportDefinitionControllerInterface
```

Comment:
May I ask what's the usecase for an interface without any methods?

Reply:
Interfaces in the PHP API client does not just define the common contract for classes that implements an interface. They also define the expected API endpoint that an implementation should communicate with if it implements the interface. Thanks for this, when you type-hint in a consumer app for an interface, you basically also type-hint to the expected API endpoint as well. (You do not care how a concreate implementation communicates with the API until it communicates with the expected API endpoint and it returns API response in the expected format.)

In this particular case, it depends on the caller (the context) whether or not the implementation of ReportDefinitionControllerInterface is interchangeable with an implementation of a CompanyReportDefinitionControllerInterface or a DeveloperReportDefinitionControllerInterface.

Reply:
Oh yeah so you use interfaces sort of a bit like a type alias in a algebratic data type. That makes sense.